File: DSA/SakshamDwivedi_590013147/Day20/D20P1.py
#Solution Function
def sort_marks(arr):
    for i in range(len(arr) - 1):
        for j in range(i, len(arr) - 1):
            if arr[j][1] < arr[j+1][1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
    return arr


#Solution Function 2
def sort_marks_2(arr):
    def grade(stu):
        return stu[1]
    arr = sorted(arr, key=grade, reverse=True)
    return arr

#Fixed Input
arr = [("Ram", 85), ("Bhuvi", 92), ("Chetan", 78)]
arr1 = [("Jatin", 88), ("Sarah", 95), ("Mona", 82), ("Aadi", 95)]
print(sort_marks(arr))
print(sort_marks(arr1))
print(sort_marks_2(arr))
print(sort_marks_2(arr1))

#Custom Input
n = int(input("Enter number of students: "))
arr = []

for i in range(n):
    name = input("Enter student name: ")
    mark = int(input("Enter marks: "))
    arr.append((name, mark))

print(sort_marks(arr))
print(sort_marks_2(arr))

File: DSA/SakshamDwivedi_590013147/Day20/D20P1.c
#include <stdio.h>
#include <string.h>

int size;
struct student{
    char name[50];
    int marks;
};

struct student arr1[100];

int main(){
    printf("Enter the number of students : ");
    scanf("%d", &size);
    getchar();
    for(int i = 0; i < size; i++){
        printf("Enter the name of the student : ");
        fgets(arr1[i].name,50,stdin);
        if(arr1[i].name[strlen(arr1[i].name) - 1] == '\n') {
            arr1[i].name[strlen(arr1[i].name) - 1] = '\0';
        }
        printf("Enter the marks of the student : ");
        scanf("%d", &arr1[i].marks);
        while(getchar() != '\n');
    }

    for(int i = 0; i < size - 1; i++){
        for(int j = 0; j < size - 1 - i; j++){
            if(arr1[j].marks < arr1[j+1].marks){
                struct student temp = arr1[j];
                arr1[j] = arr1[j+1];
                arr1[j+1] = temp;
            }
        }
    }

    for(int i = 0; i < size; i++){
        printf("%s : %d\n", arr1[i].name, arr1[i].marks);
    }

}

